src/main/scala/org/broadinstitute/dsde/rawls/ws/PerRequest.scala

```diff
@@ -36,20 +33,31 @@ trait PerRequest extends Actor with Json4sSupport {
   target ! message
 
   def receive = {
-    case RequestComplete(statusCode, response) => complete(statusCode, response)
-    case ReceiveTimeout => complete(GatewayTimeout, "Request timeout")
-    case response: AnyRef => complete(OK, response)
+    case RequestCompleteOK_(response, marshaller) => complete(OK, response)(marshaller)
+    case RequestComplete_(response, statusCode, marshaller) => complete(statusCode, response)(marshaller)
+    case RequestCompleteNoContent(statusCode) => complete(statusCode)
+    case ReceiveTimeout => complete(GatewayTimeout)
+    case x => {
+      system.log.error("Unsupported response message sent to PreRequest actor: " + Option(x).getOrElse("null").toString)
+      complete(InternalServerError)
+    }
+  }
+
+  def complete[T](status: StatusCode, obj: T)(implicit marshaller: ToResponseMarshaller[T]) = {
```

Comment:
Can you add comments on what the difference between the two complete()s are and why one would choose one or the other?